### PR TITLE
Add option for not mounting the KV mount in --dev

### PR DESCRIPTION
**Pull request by luke-clifton**
The diff is far more readable when you use the "hide whitespace" option in GitHub UI...

### CODE REVIEW

**Comment by hsimon-hashicorp** [comment]
Hi there! Please don't forget to add a [changelog entry](https://github.com/hashicorp/vault/blob/main/CONTRIBUTING.md#changelog-entries). :)

**Comment by luke-clifton** [comment]
@hsimon-hashicorp, done!

File: command/server.go [hunk 1]
```diff
@@ -2105,29 +2113,31 @@ func (c *ServerCommand) enableDev(core *vault.Core, coreConfig *vault.CoreConfig
...
-	}
-	if resp.IsError() {
-		return nil, fmt.Errorf("failed to create default KV store: %w", resp.Error())
+		}
+    resp, err := core.HandleRequest(ctx, req)
+    if err != nil {
+      return nil, fmt.Errorf("error creating default KV store: %w", err)
+    }
+    if resp.IsError() {
+      return nil, fmt.Errorf("failed to create default KV store: %w", resp.Error())
```
Comment (VioletHynes) [inline command/server.go:2139]:
```suggestion
		resp, err := core.HandleRequest(ctx, req)
		if err != nil {
		  return nil, fmt.Errorf("error creating default KV store: %w", err)
		}
		if resp.IsError() {
		  return nil, fmt.Errorf("failed to create default KV store: %w", resp.Error())
```

File: changelog/16974.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:feature
```
Comment (VioletHynes) [inline changelog/16974.txt:1]:
```suggestion
```release-note:improvement
```

**Review by VioletHynes — APPROVED** [review]
Hey! Thanks for this, this looks great. I've fixed up the merge conflicts (I think I caused them with a change I made, so it made sense :P) and I'm going to try and get this merged once everything passes.

Sorry this took so long to get some eyes on, but rest assured I'll see this to the finish line!

File: command/server.go [hunk 3]
```diff
@@ -2105,29 +2113,31 @@ func (c *ServerCommand) enableDev(core *vault.Core, coreConfig *vault.CoreConfig
...
-	}
-	if resp.IsError() {
-		return nil, fmt.Errorf("failed to create default KV store: %w", resp.Error())
+		}
+		resp, err := core.HandleRequest(ctx, req)
+		if err != nil {
+		  return nil, fmt.Errorf("error creating default KV store: %w", err)
+		}
+		if resp.IsError() {
+		  return nil, fmt.Errorf("failed to create default KV store: %w", resp.Error())
```
Comment (VioletHynes) [inline command/server.go:2139]:
```suggestion
			return nil, fmt.Errorf("error creating default KV store: %w", err)
		}
		if resp.IsError() {
			return nil, fmt.Errorf("failed to create default KV store: %w", resp.Error())
```

**Pull request merged**